Content.Server/Nyanotrasen/Weapons/Melee/MeleeStaminaCostSystem.cs

```diff
@@ -32,10 +32,21 @@ private void OnMeleeHit(EntityUid uid, MeleeStaminaCostComponent component, Mele
             if (TryComp(uid, out WieldableComponent? wieldableComponent) && wieldableComponent.Wielded)
                 coefficient = component.WieldCoefficient;
 
+            // We can see if it's a swing or a hit by checking if there are any HitEntities.
+            float staminaDamage = coefficient * (component.SwingCost + (args.HitEntities.Any() ? component.HitCost : 0f));
+            float meleeCostLimit = staminaComponent.CritThreshold * staminaComponent.MeleeCostLimitFactor;
+
+            // Avoid GetStaminaDamage: it uses some time-based prediction calculation.
+            if (staminaComponent.StaminaDamage + staminaDamage > meleeCostLimit)
+                staminaDamage = meleeCostLimit - staminaComponent.StaminaDamage;
+
+            const float ImpermissibleLimit = 0.01f;
+            if (staminaDamage <= ImpermissibleLimit)
```

Comment:
what is this check for? you can probably avoid it with a Math.Clamp for the above damage

Reply:
It's for efficiency's sake of avoiding the call to TakeStaminaDamage.

Reply:
alright